platform-sdk/docs/proposals/TSS-Block-Signing/tss-block-signing.md

```diff
@@ -0,0 +1,309 @@
+6. The `TSS-Ledger-New-Networks` proposal covers the process of setting up a new network with a ledger id.
+
+This proposal, `TSS-Block-Signing` covers the following:
+
+- `TssBaseService` API
+  - Add a method for requesting ledger signatures on a message.
+  - Add a method for registering consumers of ledger signatures.
+- `TssBaseService` Implementation
+  - Each node gossips a `TssShareSignatureTransaction` for each share signature it produces.
+  - A threshold number of `TssShareSignatureTransaction` are aggregated to produce the ledger signature.
+
+### Goals
+
+The following are goals of this proposal:
+
+1. That existing networks are able to create ledger signatures if the active roster has key material that can
+   recover the network's ledger id.  (See `TSS-Ledger-ID`)
+
+### Non-Goals
+
+The following is not part of the scope of this proposal:
+
+1. The ability to create ledger signatures for new networks.
+2. Integrating this capability into the broader block-stream effort.
+
+### Dependencies, Interactions, and Implications
+
+Dependencies on `TSS-Library`:
+
+- Upon availability of the implementation:
+  - This proposal can be tested with a pre-rendered set of shares, ledger private key, and ledger id.
+
+Dependencies on `TSS-Ledger-ID`:
+
+- Upon availability of the API:
+  - The `TssBaseService` API can be extended by this proposal.
+- Upon availability of the implementation:
+  - This proposal's full behavior can be tested in integration tests.
+
+Impacts to the Services Team:
+
+- The ability to sign blocks with TSS is blocked until this proposal delivers its capability.
+
+Implication Of Completion:
+
+- Signing blocks and construction of block proofs with TSS signatures is unblocked.
+
+### Requirements
+
+The core requirements governing the TSS proposals are stated in `TSS-Ledger-ID`.
+
+The following Block Signing specific requirements come from the Block Stream effort:
+
+1. A block is considered signed if its root hash has a ledger signature or a following block's root hash has a
+   ledger signature.
+2. TSS ledger signatures are best-effort with low probability of failure.
+3. The block proofs produced by different nodes for the same block do not need to be identical.
+4. Nodes attempt to sign every block.
+5. The number of consecutive blocks that are not signed is minimized.
+
+The rationale for these requirements is that we can
+construct [Merkle Proofs](https://ethereum.org/en/developers/tutorials/merkle-proofs-for-offline-data-integrity/) to
+extend the signature on Block N to Blocks N-1, N-2, ... all the way back to Block 1. Each block contains the root
+hash of the previous block as its first element with 1 internal merkle node between the root hash of a block and
+the root hash of the previous block. The length of a Merkle Proof extending the signature from Block N to Block
+(N-Y) is 3Y+1 hashes.
+
+To construct a TSS ledger signature, at least 1/2 the shares must sign the same message. These share signatures
```

Comment:
Here, you explicitly say 1/2, but elsewhere you discuss a configurable threshold

Reply:
You are correct.  In the TSS-Ledger-ID is it explicitly `TS +2/2` which is > 1/2 and not configuration dependent.   It is dependent on the number of shares in the candidate roster.  I will update the text for greater clarity.